src/spec.md

```diff
@@ -0,0 +1,237 @@
+
+# EthStorage SDK Interface Specification
+
+## Table of Contents
+
+- [1. Introduction](#Introduction)
+- [2. Terminology](#Terminology)
+- [3. Class Overview](#Class_Overview)
+- [4. EthStorage Class](#EthStorage)
+    - Constructor
+    - Methods
+        - read
+        - write
+- [5. FlatDirectory Class](#FlatDirectory)
+    - Constructor
+    - Methods
+        - upload
+        - download
+        - deploy
+        - setDefault
+- [6. Version History](#Version)
+
+---
+
+<p id="Introduction"></p>
+
+## 1. Introduction
+
+This document aims to standardize the interaction between applications and the EthStorage network to achieve reliable and efficient data management functionality.
+
+This document includes two main classes: `EthStorage` and `FlatDirectory`. 
+The `EthStorage` class provides asynchronous read and write operations for key-value pairs of a specified size. 
+The `FlatDirectory` class is a higher-level data management tool that provides methods for uploading and downloading data of arbitrary size.
+
+
+<p id="Terminology"></p>
+
+## 2. Terminology
+
+- **SDK**: Software Development Kit, a library that provides data upload and storage functionality.
+- **Method**: Callable functions provided by the SDK.
+- **Callback Function**: Provided by the user, passed as a parameter to methods, used to handle asynchronous operation results.
+
+
+<p id="Class_Overview"></p>
+
+## 3. Class Overview
+
+### EthStorage Class
+
+| Method Name  | Description                      |
+|--------------|----------------------------------|
+| constructor  | Create an instance of EthStorage |
+| write        | Asynchronously write data        |
+| read         | Asynchronously read data         |
+
+### FlatDirectory Class
+
+| Method Name | Description                                  |
+|-------------|----------------------------------------------|
+| constructor | Create an instance of FlatDirectory          |
+| upload      | Asynchronously upload data of arbitrary size |
+| download    | Asynchronously download data                 |
+| deploy      | Deploy a FlatDirectory contract              |
+| setDefault  | Set the default file for FlatDirectory.      |
+
+
+<p id="EthStorage"></p>
+
+## 4. EthStorage Class
+
+### Constructor
+
+**Description**: Create an instance of `EthStorage`.
+
+**Parameters**
+- `config` (object): Configuration object containing necessary settings.
+   - `rpc` (string): RPC for any evm network.
+   - `privateKey` (string): Wallet private key.
+   - `address` (string): EthStorage contract address.
```

Comment:
Is it optional because we know the default contract address for a specified RPC?